### PR TITLE
[1LP][RFR] Initial implementation for Collections control

### PR DESCRIPTION
**Pull request by quarckster**
Intent
=================

Collections conversion of cfme/control/explorer

{{pytest: -v -k "test_basic or test_bugs or test_actions or test_alert or test_instance_analysis or test_events or test_action_run_ansible_playbook or test_control_crud_ansible_playbook_action or test_control_add_ansible_playbook_action_invalid_address" --long-running --use-provider vsphere6-nested}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit eb16930420ddd0c75fb400d1a4226ed7bd1056ef:

`cfme/control/explorer/policy_profiles.py`:
- :heavy_exclamation_mark: Line 182:48: **W291** *trailing whitespace*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit bd2b5ace06a51609200173a661b7d6e8d292cf50:

`cfme/control/explorer/policy_profiles.py`:
- :heavy_exclamation_mark: Line 182:48: **W291** *trailing whitespace*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit e70f7bc21d54fdd3c2d9896663d2cb2f03e48256:

`cfme/tests/control/test_basic.py`:
- :bangbang: Line 253:1: **E302** *expected 2 blank lines, found 1*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 390a1b7d7b5ab3bbb79bf1c37f1c737efc141ea7:

`cfme/control/explorer/policies.py`:
- :red_circle: Line 11:1: **F401** *'actions.ActionCollection' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 966b4529387f2ff73eff683286f53504185d9a7b:

`cfme/control/explorer/policies.py`:
- :red_circle: Line 11:1: **F401** *'actions.ActionCollection' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 58685e8dfbb6aefa977966060e2dbf21941d5a80:

`cfme/tests/ansible/test_embedded_ansible_actions.py`:
- :red_circle: Line 58:26: **F821** *undefined name 'policy_profiles'*
- :red_circle: Line 63:26: **F821** *undefined name 'policies'*
- :red_circle: Line 68:26: **F821** *undefined name 'conditions'*
- :red_circle: Line 73:26: **F821** *undefined name 'actions'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/control/explorer/actions.py [hunk 1]
```diff
@@ -112,7 +112,47 @@ def is_displayed(self):
...
+
+    def __init__(self, appliance):
+        self.appliance = appliance
+
+    def instantiate(self, description, action_type, action_values=None):
+        return Action(self, description, action_type, action_values=action_values)
+
+    def create(self, description, action_type, action_values=None):
+        """Create an Action in the UI."""
+        action = self.instantiate(description, action_type, action_values=action_values)
```
Comment (mfalesni) [inline cfme/control/explorer/actions.py]: If action_values keys and values correspond to the view, then you should be able to move this to the end of this method, which would be preferable.

File: cfme/control/explorer/policy_profiles.py [hunk 2]
```diff
@@ -157,30 +158,40 @@ def exists(self):
...
+@navigator.register(PolicyProfileCollection, "Add")
 class PolicyProfileNew(CFMENavigateStep):
     VIEW = NewPolicyProfileView
-    prerequisite = NavigateToAttribute("appliance.server", "ControlExplorer")
+    prerequisite = NavigateToSibling("All")
 
     def step(self):
-        self.view.policy_profiles.tree.click_path("All Policy Profiles")
-        self.view.configuration.item_select("Add a New Policy Profile")
+        self.prerequisite_view.configuration.item_select("Add a New Policy Profile")
```
Comment (mfalesni) [inline cfme/control/explorer/policy_profiles.py]: :+1:

File: cfme/tests/control/test_actions.py [hunk 3]
```diff
@@ -212,37 +212,49 @@ def policy_profile_name(name_suffix):
...
 @pytest.fixture(scope="module")
-def compliance_condition():
-    condition = conditions.VMCondition(
+def action_collection(appliance):
+    return appliance.get(actions.ActionCollection)
+
+
+@pytest.fixture(scope="module")
+def compliance_condition(appliance):
+    condition_collection = appliance.get(conditions.ConditionCollection)
```
Comment (mfalesni) [inline cfme/tests/control/test_actions.py]: Shouldn't you use something like this in the other test files?

**Pull request merged**